s/start.go

```diff
@@ -111,8 +179,34 @@ func (spec SupervisorSpec) start(parentCtx context.Context, parentName string) (
 		},
 	}
 
+	onStart := func(err startError) {
+		if err != nil {
+			startCh <- err
+		}
+		close(startCh)
+	}
+
+	onTerminate := func(err terminateError) {
+		if err != nil {
+			terminateCh <- err
+		}
+		close(terminateCh)
```

Comment:
Same comment as above.

Reply:
Same ordeal, the child termination happens in order and we report all the ones that failed in one go, so this function gets called only once per supervisor